### PR TITLE
kotlin-dsl support

### PR DESCRIPTION
**Pull request by drieks**
Hi @ftomassetti,

I have finished the kotlin-dsl support. Please let me know if anything is unclear or missing..

### CODE REVIEW

File: antlr-kotlin-examples-jvm/.gitignore [hunk 1]
```diff
@@ -1 +0,0 @@
-generated-src
```
Comment (ftomassetti) [inline antlr-kotlin-examples-jvm/.gitignore]: why?

File: antlr-kotlin-examples-jvm/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+*/
+
+tasks.generateKotlinGrammarSource {
+    maxHeapSize = "64m"
+    packageName = "com.strumenta.antlrkotlin.examples"
+}
+
+/*
+// use this settings if you want to add the generated sources to version control
+tasks.generateKotlinGrammarSource {
```
Comment (ftomassetti) [inline antlr-kotlin-examples-jvm/build.gradle.kts]: only the second part, not this one. This one is needed, right?

File: antlr-kotlin-examples-jvm/gradle [hunk 3]
```diff
@@ -0,0 +1 @@
+../gradle
```
Comment (ftomassetti) [inline antlr-kotlin-examples-jvm/gradle]: should this be gradlew?

File: antlr-kotlin-examples-jvm/gradle.properties [hunk 4]
```diff
@@ -1,5 +0,0 @@
-org.gradle.jvmargs=-Xmx8192m
```
Comment (ftomassetti) [inline antlr-kotlin-examples-jvm/gradle.properties]: don't we need more memory to build this?

File: antlr-kotlin-examples-jvm/src/test/kotlin/com/strumenta/antlrkotlin/examples/MiscStringTest.kt [hunk 5]
```diff
@@ -11,19 +14,21 @@ fun assertArrayEquals(a: CharArray, b: CharArray) {
 class MiscStringTest {
 
 //    @Test fun testToCharArrayEmpty() {
-//        assertArrayEquals(charArrayOf(), "".toCharArray())
+//        com.strumenta.antlrkotlin.examples.assertArrayEquals(charArrayOf(), "".toCharArray())
```
Comment (ftomassetti) [inline antlr-kotlin-examples-jvm/src/test/kotlin/com/strumenta/antlrkotlin/examples/MiscStringTest.kt]: mmm, this seems weird

File: antlr-kotlin-examples-jvm/src/test/kotlin/com/strumenta/antlrkotlin/examples/MiscStringTest.kt [hunk 6]
```diff
@@ -11,19 +14,21 @@ fun assertArrayEquals(a: CharArray, b: CharArray) {
...
 
-    @Test fun testToCharArrayEmptyEl0() {
+    @Test
+    fun testToCharArrayEmptyEl0() {
         assertEquals('a', "abc def".toCharArray()[0])
     }
 
 //    @Test fun testToCharArray() {
-//        assertArrayEquals(charArrayOf('a', 'b', 'c', ' ', 'd', 'e', 'f'), "abc def".toCharArray())
+//        com.strumenta.antlrkotlin.examples.assertArrayEquals(charArrayOf('a', 'b', 'c', ' ', 'd', 'e', 'f'), "abc def".toCharArray())
```
Comment (ftomassetti) [inline antlr-kotlin-examples-jvm/src/test/kotlin/com/strumenta/antlrkotlin/examples/MiscStringTest.kt]: also this

File: antlr-kotlin-gradle-plugin/src/main/java/com/strumenta/antlrkotlin/gradleplugin/AntlrKotlinPluginExtension.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 the original author or authors.
```
Comment (ftomassetti) [inline antlr-kotlin-gradle-plugin/src/main/java/com/strumenta/antlrkotlin/gradleplugin/AntlrKotlinPluginExtension.java]: We should probably update this :D

File: antlr-kotlin-examples-jvm/.gitignore (hunk 1 above)
Comment (drieks) [reply antlr-kotlin-examples-jvm/.gitignore]: the gradle default is to put this into "antlr-kotlin-examples-jvm/build/generated-src/antlr/main", the build folder is already ignored by git

File: antlr-kotlin-examples-jvm/build.gradle.kts (hunk 2 above)
Comment (drieks) [reply antlr-kotlin-examples-jvm/build.gradle.kts]: I have added more comments

File: antlr-kotlin-examples-jvm/gradle (hunk 3 above)
Comment (drieks) [reply antlr-kotlin-examples-jvm/gradle]: I symlinked the gradle directory, gradlew and gradlew.bat to the parent gradlew installation, so we just need to update only one place.

File: antlr-kotlin-examples-jvm/gradle.properties (hunk 4 above)
Comment (drieks) [reply antlr-kotlin-examples-jvm/gradle.properties]: I just removed all old gradle project files.. it is still compiling. I think it is not required.

File: antlr-kotlin-examples-jvm/src/test/kotlin/com/strumenta/antlrkotlin/examples/MiscStringTest.kt (hunk 5 above)
Comment (drieks) [reply antlr-kotlin-examples-jvm/src/test/kotlin/com/strumenta/antlrkotlin/examples/MiscStringTest.kt]: I think this is a search&replace mistake

File: antlr-kotlin-examples-jvm/src/test/kotlin/com/strumenta/antlrkotlin/examples/MiscStringTest.kt (hunk 6 above)
Comment (drieks) [reply antlr-kotlin-examples-jvm/src/test/kotlin/com/strumenta/antlrkotlin/examples/MiscStringTest.kt]: fixed

**Pull request merged**